docs/madr/decisions/049-policy-on-namespace.md

```diff
@@ -0,0 +1,529 @@
+# Policy On Namespace
+
+- Status: accepted
+
+## Context and Problem Statement
+
+Kuma 2.6.0 enable users to apply policies on Zone CP.
+This change unblocks us to introduce namespace support for policies.
+The CRDs for new policies are already namespace-scoped, but they can be applied only on the `kuma-system` namespace.
+
+## Considered Options
+
+- allow applying new policies to the custom namespaces
+- do nothing
+
+## Decision Outcome
+
+- allow applying new policies to the custom namespaces
+
+## Pros and Cons of the applying policies on custom namespaces
+
+* Good, because it allows using Kubernetes RBAC
+* Good, because it provides more predictable behaviour as namespace-scoped policy affects only workloads in the same
+  namespace
+* Good, because it provides more Kubernetes-native UX
+* Bad, because adds complexity for users
+    * what's the right namespace for the policy?
+    * how cross-policy refs work?
+    * how policy order works?
+
+## Pros and Cons of doing nothing
+
+* Good, because we have time to work on other features
+* Bad, because cluster operator has to grant write permissions to `kuma-system` namespace to everyone who works with
+  policies
+* Bad, because low isolation between teams (team-a can unintentionally break polices of the team-b)
+
+## Implementation
+
+Applying a policy on the custom namespace **affects requests to the pods in the custom namespace**.
+Since some policies are applied on the client-side, the scope of policy's effect is not limited to the namespace
+where it was applied. 
+For example, the following policy sets the timeout on outbounds of all consumers of the backend service:
+
+```yaml
+kind: MeshTimeout
+metadata:
+  namespace: backend-ns
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef: 
+        kind: MeshService
+        name: backend
+        namespace: backend-ns
+      default:
+        connectionTimeout: 5s
+```
+
+Important to note, even though policy in `backend-ns` changes proxy configurations in `frontend-ns` 
+the top-level targetRef still servers the purpose of showing what proxy configurations are going to change.
+Policy's namespace **does not** implicitly turn the top-level targetRef to `MeshSubset{tags:{"kuma.io/namespace":"backend-ns"}}`.
+
+### Namespace-scoped policies and multizone
+
+Namespace-scoped policies can be applied to a custom namespace only on zones as there are simply no DPPs and custom
+namespaces on Global.
+
+After a user applied a policy on zone's custom namespace, the webhook automatically adds a `k8s.kuma.io/namespace` label to the policy.
+The label is part of the hash suffix when policy is synced to global.
+
+As mentioned previously, a policy applied in producer namespace can affect envoy configs of consumers.
```

Comment:
This is the first ocurence of the work "producer" so it's not "mentioned previously"